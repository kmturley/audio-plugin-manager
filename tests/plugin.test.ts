import { beforeAll, expect, test } from 'vitest';
import { cli, CliOutput } from './shared';
import path from 'path';
import { dirDelete } from '@studiorack/core';

const PLUGIN_DIR: string = path.join('test', 'plugins');
const PLUGIN_ID: string = 'studiorack/mda/mda';

beforeAll(async () => {
  await cli(`config set pluginFolder "${PLUGIN_DIR}"`);
  dirDelete(PLUGIN_DIR);
});

test('Plugin get', async () => {
  const output: CliOutput = await cli(`plugin get "${PLUGIN_ID}" --json`);
  expect(output.exitCode).toBe(0);
  expect(output.stdout).toMatchSnapshot();
});

test('Plugin install', async () => {
  const output: CliOutput = await cli(`plugin install "${PLUGIN_ID}" --json`);
  expect(output.exitCode).toBe(0);
  expect(output.stdout).toMatchSnapshot();
});
