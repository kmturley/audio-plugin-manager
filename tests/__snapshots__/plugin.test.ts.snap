// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Plugin get 1`] = `
"{
  "author": "Paul Kellett",
  "homepage": "http://mda.smartelectronix.com",
  "name": "MDA",
  "description": "Collection of effect plug-ins, from delay to an overdrive and a vocoder.",
  "tags": [
    "Effect",
    "Delay",
    "Vocoder"
  ],
  "date": "2020-12-20T08:00:00.000Z",
  "files": {
    "audio": {
      "name": "mda.flac",
      "size": 127392,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda.flac"
    },
    "image": {
      "name": "mda.jpg",
      "size": 104156,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda.jpg"
    },
    "linux": {
      "name": "mda-linux.zip",
      "size": 70938,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda-linux.zip"
    },
    "mac": {
      "name": "mda-mac.zip",
      "size": 5122516,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda-mac.zip"
    },
    "win": {
      "name": "mda-win.zip",
      "size": 1247768,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda-win.zip"
    }
  },
  "license": "gpl-3.0",
  "id": "studiorack/mda/mda",
  "version": "1.0.4"
}
"
`;

exports[`Plugin install 1`] = `
"{
  "author": "Paul Kellett",
  "homepage": "http://mda.smartelectronix.com",
  "name": "MDA",
  "description": "Collection of effect plug-ins, from delay to an overdrive and a vocoder.",
  "tags": [
    "Effect",
    "Delay",
    "Vocoder"
  ],
  "date": "2020-12-20T08:00:00.000Z",
  "files": {
    "audio": {
      "name": "mda.flac",
      "size": 127392,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda.flac"
    },
    "image": {
      "name": "mda.jpg",
      "size": 104156,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda.jpg"
    },
    "linux": {
      "name": "mda-linux.zip",
      "size": 70938,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda-linux.zip"
    },
    "mac": {
      "name": "mda-mac.zip",
      "size": 5122516,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda-mac.zip"
    },
    "win": {
      "name": "mda-win.zip",
      "size": 1247768,
      "url": "https://github.com/studiorack/mda/releases/download/v1.0.4/mda-win.zip"
    }
  },
  "license": "gpl-3.0",
  "id": "studiorack/mda/mda",
  "version": "1.0.4",
  "paths": [
    "test/plugins/Components/studiorack/mda/mda/1.0.4/mda.component",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Ambience.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Bandisto.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda BeatBox.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Combo.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda DX10.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda De-ess.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Degrade.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Delay.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Detune.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Dither.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda DubDelay.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Dynamics.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Image.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda JX10.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Leslie.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Limiter.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Looplex.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Loudness.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda MultiBand.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Overdrive.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Piano.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda RePsycho!.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda RezFilter.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda RingMod.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda RoundPan.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Shepard.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Splitter.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Stereo.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda SubBass.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Talkbox.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda TestTone.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda ThruZero.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Tracker.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda VocInput.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda Vocoder.vst",
    "test/plugins/VST/studiorack/mda/mda/1.0.4/mda ePiano.vst"
  ],
  "status": "installed"
}
"
`;
